jax/_src/lax/lax.py

```diff
@@ -6083,6 +6083,67 @@ def _rng_uniform_translation_rule(c, a, b, *, shape):
 xla.translations[rng_uniform_p] = _rng_uniform_translation_rule
 
 
+def _rng_bit_generator_shape_rule(key, *, shape, dtype, algorithm):
+  _ = dtype, algorithm
+  return (key.shape, tuple(shape))
+
+
+def _rng_bit_generator_dtype_rule(key, *, shape, dtype, algorithm):
+  _ = key, shape, algorithm
+  return (key.dtype, dtype)
+
+
+def _rng_bit_generator_weak_type_rule(key, *, shape, dtype, algorithm):
+  _ = shape, dtype, algorithm
+  return (key.weak_type, False)
+
+
+def _rng_bit_generator_translation_rule(c, key, *, shape, dtype, algorithm):
+  _ = c
+  xla_shape = xc.Shape.array_shape(np.dtype(dtype), shape)
+  return xops.RngBitGenerator(algorithm, key, xla_shape)
+
+
+def _rng_bit_generator_named_shape_rule(key, *, shape, dtype, algorithm):
+  return [key.named_shape, key.named_shape]
+
+rng_bit_generator_p = Primitive("rng_bit_generator")
+rng_bit_generator_p.multiple_results = True
+rng_bit_generator_p.def_impl(partial(xla.apply_primitive, rng_bit_generator_p))
+rng_bit_generator_p.def_abstract_eval(
+    partial(standard_multi_result_abstract_eval, rng_bit_generator_p,
+            _rng_bit_generator_shape_rule, _rng_bit_generator_dtype_rule,
+            _rng_bit_generator_weak_type_rule,
+            _rng_bit_generator_named_shape_rule))
+xla.translations[rng_bit_generator_p] = _rng_bit_generator_translation_rule
+
+RandomAlgorithm = xops.RandomAlgorithm
```

Comment:
Actually I guess you would need to gate these with a version check. I think it's no big deal though:
```
if jax.lib.version >= (...):
  RandomAlgorithm = ....
```

Reply:
I think I managed to condition all of the logic on jax.lib.version. It is not the prettiest but should be easy to read and we can get rid of the extra complexity when the minimum jaxlib version is bumped